src/components/list/demoListControls/index.html

```diff
@@ -0,0 +1,29 @@
+<md-list ng-controller="ListCtrl">
+  <md-subheader class="md-no-sticky">Single Action Checkboxes</md-subheader>
+  <md-item ng-repeat="topping in toppings">
+    <p> {{ topping.name }} </p>
+    <md-checkbox class="md-secondary" ng-model="topping.wanted" aria-label="{{ topping.name }}">
+  </md-item>
+  <md-subheader class="md-no-sticky">Clickable Items with Secondary Controls</md-subheader>
+  <md-item ng-click="navigateTo(setting.extraScreen)" ng-repeat="setting in settings">
+    <md-icon md-svg-icon="{{setting.icon}}"></md-icon>
+    <p> {{ setting.name }} </p>
+    <md-switch class="md-secondary" ng-model="setting.enabled" aria-label="{{ setting.name }}">
+  </md-item>
+  <md-item ng-click="navigateTo('data usage')">
+    <md-icon md-svg-icon="cached"></md-icon>
+    <p>Data Usage</p>
+  </md-item>
+  <md-subheader class="md-no-sticky">Checkbox with Secondary Action</md-subheader>
+  <md-item ng-repeat="message in messages">
+    <md-checkbox ng-model="message.selected" aria-label="Select"></md-checkbox>
```

Comment:
In the compiled code, the `<div role="button">` wrapper and checkbox with `tabIndex="-1"` don't currently work for accessibility because the user isn't focused on the control that's actually being toggled (they are focused on the button "wrapper"). 

Ideally, focus would go directly onto the checkbox and you'd toggle it there, so that its semantics were announced. The `<div role=button>` is also unnecessary around a switch, as I'm pretty sure you wouldn't ever nest a checkbox inside a button. I think any place we transform a list item with a switch as its primary action, we're going to run into this problem with the code in its current state. We *may* be able to try something tricky on the parent, like `aria-controls`, but we'll have to test it.